pywhy_stats/independence/kci.py

```diff
@@ -1,15 +1,23 @@
-from functools import partial
```

Comment:
Is it possible to separate the changes in the existing kernel tests from adding the discrepancy tests? There are now several things/modifications we should check in a separate PR. If necessary, there can be a third PR then afterwards that combines the discrepancy tests with the changes in the kernel tests.